### PR TITLE
feat(mobile): task management bb-445

### PR DESCRIPTION
**Pull request by Olha-Katran**
Add text palceholder for empty active/past tasks
Add background for task screen.

Screenshot: 

Video: 

https://github.com/user-attachments/assets/52d9b2a5-e3ca-4569-91b4-14c26b52c375

### CODE REVIEW

File: apps/mobile/src/libs/components/page-switcher/page-switcher.tsx [hunk 1]
```diff
@@ -9,22 +9,26 @@ import { styles } from "./style";
...
 	activeTab,
+	isDisabled = false,
 	onTabChange,
 	style,
 	tabs,
 }) => {
 	const handleTabPress = useCallback(
 		(tab: string): void => {
-			onTabChange(tab);
+			if (!isDisabled) {
```
Comment (revenets) [inline apps/mobile/src/libs/components/page-switcher/page-switcher.tsx:27]: Why not just pass the `isDisabled ` as TouchableOpacity prop?

File: apps/mobile/src/screens/tasks/tasks.tsx [hunk 2]
```diff
@@ -104,16 +115,50 @@ const Tasks: React.FC = () => {
...
+									style={[globalStyles.ph32, styles.empty]}
+									weight="bold"
+								>
+									{mode === TasksMode.CURRENT
+										? "You don't have any active tasks"
+										: "You don't have any completed tasks"}
+								</Text>
+							</View>
+						) : (
+							<ScrollView
```
Comment (revenets) [inline apps/mobile/src/screens/tasks/tasks.tsx:146]: I'd suggest using FlatList instead.

File: apps/mobile/src/screens/tasks/tasks.tsx [hunk 3]
```diff
@@ -36,6 +38,15 @@ const Tasks: React.FC = () => {
...
 
+	const currentTasks: TaskDto[] = tasks.filter(
+		(task) => task.status === TaskStatus.CURRENT,
+	);
+	const pastTasks: TaskDto[] = tasks.filter(
+		(task) => task.status === TaskStatus.COMPLETED,
+	);
+	const isCurrentEmpty = currentTasks.length === NumericalValue.ZERO;
+	const isPastEmpty = pastTasks.length === NumericalValue.ZERO;
+
```
Comment (revenets) [inline apps/mobile/src/screens/tasks/tasks.tsx:49]: Why do we need this? Regarding the logic below - we already have filtered tasks returned from the store.

File: apps/mobile/src/screens/tasks/tasks.tsx [hunk 4]
```diff
@@ -104,16 +115,50 @@ const Tasks: React.FC = () => {
...
+								<Text
+									preset="subheading"
+									style={[globalStyles.ph32, styles.empty]}
+									weight="bold"
+								>
+									{mode === TasksMode.CURRENT
+										? "You don't have any active tasks"
+										: "You don't have any completed tasks"}
+								</Text>
+							</View>
```
Comment (revenets) [inline apps/mobile/src/screens/tasks/tasks.tsx:144]: This can be moved to a separate component and used as a ListEmptyComponent prop of FlatList

**Review by revenets — CHANGES_REQUESTED** [review]

File: apps/mobile/src/assets/images/background.png
Comment (fshabanov) [inline apps/mobile/src/assets/images/background.png:1]: Prefer svg

File: apps/mobile/src/screens/tasks/tasks.tsx [hunk 5]
```diff
@@ -116,6 +110,34 @@ const Tasks: React.FC = () => {
...
+					onSkip={handleSkip}
+					task={item}
+				/>
+			);
+		},
+		[handleComplete, handleExpired, handleSkip],
+	);
+
+	const handleKeyExtractor = useCallback(
+		(item: TaskDto, index: number) => index.toString(),
```
Comment (revenets) [inline apps/mobile/src/screens/tasks/tasks.tsx:129]: Use` item.id` instead

**Review by revenets — APPROVED** [review]

File: apps/mobile/src/slices/task/task.slice.ts [hunk 6]
```diff
@@ -14,16 +15,18 @@ import {
...
+	currentTasks: TaskDto[];
 	dataStatus: ValueOf<typeof DataStatus>;
 	expiredTasks: TaskDto[];
-	tasks: TaskDto[];
+	pastTasks: TaskDto[];
 };
 
 const initialState: State = {
 	activeTasks: [],
+	currentTasks: [],
```
Comment (revenets) [inline apps/mobile/src/slices/task/task.slice.ts:26]: It seems to be unused. Do we still need it?

**Pull request merged**